src/main/java/com/example/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,78 @@
+package com.example.bookstore.controller;
+
+import com.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstore.dto.category.CategoryDto;
+import com.example.bookstore.dto.category.CategoryResponseDto;
+import com.example.bookstore.dto.category.CreateCategoryRequestDto;
+import com.example.bookstore.mapper.BookMapper;
+import com.example.bookstore.service.BookService;
+import com.example.bookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories management", description = "Endpoints for managing categories")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+    private final BookMapper bookMapper;
+
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
+    @Operation(summary = "Get categories", description = "Get a list of all available categories")
+    @GetMapping
+    public List<CategoryResponseDto> getAll(Pageable pageable) {
+        return categoryService.getAll(pageable);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Create category", description = "create new category")
+    @PostMapping
+    public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto categoryDto) {
+        return categoryService.createBook(categoryDto);
```

Comment:
Forgot to change the name of the method in `CategoryService` to `createCategory`